eth_account/hdaccount/deterministic.py

```diff
@@ -0,0 +1,211 @@
+        assert len(parent_key) == 32  # Should be guaranteed here in return statment
+        child = hmac_sha512(parent_chain_code, b"\x00" + parent_key + node.serialize())
+
+    else:
+        assert len(ec_point(parent_key)) == 33  # Should be guaranteed by Account class
+        child = hmac_sha512(parent_chain_code, ec_point(parent_key) + node.serialize())
+
+    assert len(child) == 64
+
+    if to_int(child[:32]) >= SECP256K1_N:
+        # Invalid key, compute using next node (< 2**-127 probability)
+        return derive_child_key(parent_key, parent_chain_code, node + 1)
+
+    child_key = (to_int(child[:32]) + to_int(parent_key)) % SECP256K1_N
+    if child_key == 0:
+        # Invalid key, compute using next node (< 2**-127 probability)
+        return derive_child_key(parent_key, parent_chain_code, node + 1)
+
+    child_key = child_key.to_bytes(32, byteorder="big")
+    child_chain_code = child[32:]
+    return child_key, child_chain_code
+
+
+class HDPath:
+    def __init__(self, path: str):
+        """
+        Constructor for this class. Initializes an hd account generator using the
+        given path string (from BIP-0032). The path is decoded into nodes of the
+        derivation key tree, which define a pathway from a given master seed to
+        the child key that is used for a given purpose. Please also reference BIP-
+        0043 (which definites the first level as the "purpose" field of an HD path)
+        and BIP-0044 (which defines a commonly-used, 5-level scheme for BIP32 paths)
+        for examples of how this object may be used. Please note however that this
+        object makes no such assumptions of the use of BIP43 or BIP44, or later BIPs.
+        :param path             : BIP32-compatible derivation path
+        :type path              : str as "m/idx_0/.../idx_n" or "m/idx_0/.../idx_n"
+                                  where idx_* is either an integer value (soft node)
+                                  or an integer value followed by either the "'" char
+                                  or the "H" char (hardened node)
+        """
+        nodes = path.split('/')
+        if nodes[0] not in ('m', 'M'):
+            raise ValueError(f'Path is not valid: "{path}". Must start with "m"')
+        decoded_path = []
+        try:
+            for node in nodes[1:]:  # We don't need the root node 'm'
+                decoded_path.append(Node.decode(node))
+        except ValueError as e:
+            raise ValueError(f'Path is not valid: "{path}". Issue with node "{node}": {e}')
+        self._path = decoded_path
+
+    def __repr__(self) -> str:
+        return f'{self.__class__.__name__}(path="{self.encode()}")'
+
+    def encode(self) -> str:
+        """
+        Encodes this class to a string (reversing the decoding in the constructor)
+        """
+        encoded_path = ['m']
+        for node in self._path:
+            encoded_path.append(node.encode())
```

Comment:
*mutability-nitpick*

Would you be ok modeling this as.  

```python
encoded_path = ('m',) + tuple(node.encode() for node in self._path)
```

Both personal preference, and wanting to avoid enshrining mutability patterns into the code for fear that it encourages more of it.  Immutable by default unless we have an explicit reason to need it is the general policy.

Reply:
Will do